gamersnet_backend/__tests__/posts/testPostUpdate.js

```diff
@@ -0,0 +1,160 @@
+    await seedDB();
+});
+
+async function seedDB() {
+
+    //insert users
+    await users.deleteMany();
+    await tokens.deleteMany();
+    await posts.deleteMany();
+
+    //user1,token 1, post 1
+    let user1Inserted = await users.insertOne(user1);
+    user1ID = ObjectId(user1Inserted.insertedId);
+
+    token1.userID = user1ID;
+    let token1Inserted = await tokens.insertOne(token1)
+
+    post1.userID = user1ID;
+    let post1Inserted = await posts.insertOne(post1);
+    post1ID = post1Inserted.insertedId;
+
+    //user2,token 2, post 2
+    let user2Inserted = await users.insertOne(user2);
+    user2ID = ObjectId(user2Inserted.insertedId);
+
+    token2.userID = user2ID;
+    let token2Inserted = await tokens.insertOne(token2)
+
+    post2.userID = user2ID;
+    let post2Inserted = await posts.insertOne(post2);
+    post2ID = post2Inserted.insertedId;
+
+}
+
+afterAll(async () => {
+    // this is important to do, otherwise the db client remains open and tests never exit
+    await MongoDB.close();
+})
+
+
+
+describe('Test Post Updates', () => {
+    test('User 1 is logged in and can update post of user 1', (done) => {
+        return request(app).post('/posts/updatePost')
+        .set('Cookie', 'token=user1_token')
+        .query({_id: post1ID.toHexString()})
+        .send({
+            description: "post by user1, (UPDATE)",
+            gameName: "xyz",
+            numPlayers: 5, //null in case of incorrect format
+            gameTimeUTC: new Date(), 
+            duration: "1hr",
+            location: "Earth"
+        })
+        .expect(201).end(done);
+    });
+
+
+    test('User 1 is logged in and cannot update post of user 2', (done) => {
+        return request(app).post('/posts/updatePost')
+        .set('Cookie', '')//clear cookie and reset
+        .set('Cookie', 'token=user1_token')//logged in as user 2
+        .query({_id: post2ID.toHexString()}) //request to update post of user 1
+        .send({
+            description: "post by user2, (SHOULD NOT UPDATE)",
+            gameName: "xyz",
+            numPlayers: 5, //null in case of incorrect format
+            gameTimeUTC: new Date(), 
+            duration: "1hr",
+            location: "Pluto"
+        })
+        .expect(401).end(done);
+    });
+
+
+    test('User 2 login session expired and cannot update post of user 2', (done) => {
+        return request(app).post('/posts/updatePost')
+        .set('Cookie', '')//clear cookie and reset
+        .set('Cookie', 'token=user2_token') //was logged as user2 but session expired
+        .query({_id: post1ID.toHexString()})
+        .send({
+            _id: post1ID,
+            userID: user1ID,// intended to link to existing users in db
+            description: "post by user2, (SHOULD NOT UPDATE)",
+            gameName: "xyz",
+            numPlayers: 5, //null in case of incorrect format
+            gameTimeUTC: new Date(), 
+            duration: "1hr",
+            location: "Pluto"
+        })
```

Comment:
Shouldn't there be one more test where the post doesn't exist in DB, but the user is trying to update it?
Just to check the status code!

Reply:
Sure, i can do that, but that would need a postID to be passed in ...like updatePost(postID), and if the post is not found, it won't update... but will return 201 anyways.. I'll add the test